server/ActiveOrderDao.js

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable promise/always-return */
+class ActiveOrdersDao {
+    constructor(gsDB) {
+        this.gsDB = gsDB;
+    }
+
+    updateActiveOrderStatus(orderId, newStatus) {
+        //get order from db using orderId
+        let orderRef = this.gsDB.collection('ActiveOrdes').doc(String(orderId));
+        //modify order status
+        return orderRef.update({
+            "status": newStatus
+        }).catch(_err => {
+            console.log("Failed to update order " + orderId)
+            // eslint-disable-next-line promise/always-return
+        }).then(() => {
+            console.log("Active order" + orderId + " status updated")
+        });
+    }
+
+    addToActiveOrders(order) {
+        //write order to db; index by orderId
+        let key = generateUniqueKey();
+        this.gsDB.ref('ActiveOrders/' + String(key)).set(order);
+    }
+
+    removeFromActiveOrders(orderId) {
+        //get all documents with orderid
+        let orderRef = this.gsDB.collection('ActiveOrdes').doc(String(orderId));
+        // eslint-disable-next-line promise/always-return
+        orderRef.get().then(snapshot => {
```

Comment:
is there a need for the forEach instead of :

db.collection("ActiveOrders").doc(String(orderId)).delete().then(function() {
    console.log("Document successfully deleted!");
}).catch(function(error) {
    console.error("Error removing document: ", error);
});